packages/url_launcher/url_launcher_web/test/url_launcher_web_test.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@TestOn('chrome') // Uses web-only Flutter SDK
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:url_launcher/url_launcher.dart';
+import 'package:url_launcher_web/url_launcher_web.dart';
+
+void main() {
+  group('URL Launcher for Web', () {
+    setUp(() {
+      TestWidgetsFlutterBinding.ensureInitialized();
+      webPluginRegistry.registerMessageHandler();
+      final Registrar registrar = webPluginRegistry.registrarFor(UrlLauncherPlugin);
+      UrlLauncherPlugin.registerWith(registrar);
+    });
+
+    test('can launch "http" URLs', () {
+      expect(canLaunch('http://google.com'), completion(isTrue));
+    });
+
+    test('can launch "https" URLs', () {
+      expect(canLaunch('https://google.com'), completion(isTrue));
+    });
+
+    test('cannot launch "tel" URLs', () {
+      expect(canLaunch('tel:5551234567'), completion(isFalse));
+    });
+  });
```

Comment:
is there a way to test launch ?

Reply:
It would require a lot of extra code. Rather than running the test using `flutter test --platform chrome` we would have to spin up our own instance of Chrome, somehow build the JS getting the Flutter Web SDK from somewhere, call `launch`, and determine that a new tab was opened with the correct API.

Reply:
I added some basic tests for `launch`